### PR TITLE
[12.0][MIG] hr_holidays_validity_date

### PR DESCRIPTION
**Pull request by jarroyomorales**
Migration to 12.0 of `hr_holidays_validity_date`.

Odoo already adds this feature in 12.0 but simply adds it as a constraint. I removed that part and just kept the part that handles whether it should display an error or a warning, which I think it's interesting too. 
I also added a migration script that converts this module old fields to the new fields Odoo added in leaves 12.0.

### CODE REVIEW

**Review by etobella — APPROVED** [review]
LGTM. Maybe you should change of behaviour on the HISTORY. WDYT?

**Comment by jarroyomorales** [comment]
@etobella Rebased to fix travis and added HISTORY.rst

**Comment by Saran440** [comment]
Please check this error when create holidays leave
`TypeError: '<' not supported between instances of 'datetime.date' and 'bool'`
but It's workable.

**Comment by jarroyomorales** [comment]
@saran440 Fixed!

File: hr_holidays_validity_date/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{
+    'name': "HR holidays validity date",
+
+    'summary': """
+        Allow to define start and end date on holidays type.""",
+    'author': 'ACSONE SA/NV,'
+              'Odoo Community Association (OCA)',
+    'website': "https://github.com/OCA/hr",
+    'category': 'Human Resources',
+    'version': '12.0.2.0.0',
```
Comment (Saran440) [inline hr_holidays_validity_date/__manifest__.py]:
```suggestion
    'version': '12.0.1.0.0',
```
First version in V12 should be `12.0.1.0.0`

File: hr_holidays_validity_date/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    'license': 'AGPL-3',
+    'depends': [
+        'hr_holidays',
+    ],
+    'data': [
```
Comment (Saran440) [inline hr_holidays_validity_date/__manifest__.py:16]: installable attribute missing in manifest

File: hr_holidays_validity_date/models/hr_leave.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        compute='_compute_warning_range',
+        oldname='warning',
+    )
+    restrict_dates = fields.Boolean(string='Restrict dates',
+                                    related='holiday_status_id.restrict_dates')
+
+    @api.depends('holiday_status_id', 'date_from', 'date_to')
+    def _compute_warning_range(self):
+        for rec in self:
+            if rec.holiday_status_id.validity_start and (
```
Comment (Saran440) [inline hr_holidays_validity_date/models/hr_leave.py:31]: Create variable and check if 1 time better?
>  if vstart and vstop and dfrom and dto and ( dfrom.date() < vstart or dto.date() > vstop):

**Review by Saran440 — COMMENTED** [review]
Code review.

File: hr_holidays_validity_date/__manifest__.py (hunk 1 above)
Comment (jarroyomorales) [reply hr_holidays_validity_date/__manifest__.py]: Changed!

File: hr_holidays_validity_date/__manifest__.py (hunk 2 above)
Comment (jarroyomorales) [reply hr_holidays_validity_date/__manifest__.py:16]: Added it, but is it mandatory?

File: hr_holidays_validity_date/models/hr_leave.py (hunk 3 above)
Comment (jarroyomorales) [reply hr_holidays_validity_date/models/hr_leave.py:31]: I don't see where I can reuse the if, in every iteration a record may have a different leave type or different attributes in general

File: hr_holidays_validity_date/__manifest__.py (hunk 2 above)
Comment (nikul-serpentcs) [reply hr_holidays_validity_date/__manifest__.py:16]: @jarroyomorales FYI https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#manifest

File: hr_holidays_validity_date/tests/test_hr_holidays_validity_date.py [hunk 4]
```diff
@@ -1,27 +1,3 @@
-##############################################################################
```
Comment (nikul-serpentcs) [inline hr_holidays_validity_date/tests/test_hr_holidays_validity_date.py]: @jarroyomorales here version 11 copyright:  `Copyright (c) 2015 ACSONE SA/NV (<http://acsone.eu>)`

File: hr_holidays_validity_date/tests/test_hr_holidays_validity_date.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
+from odoo.tests import common
```
Comment (nikul-serpentcs) [inline hr_holidays_validity_date/tests/test_hr_holidays_validity_date.py:4]: @jarroyomorales in v12: Copyright is missing

**Comment by jarroyomorales** [comment]
@nikul-serpentcs Comments attended

File: hr_holidays_validity_date/models/hr_leave.py (hunk 3 above)
Comment (Saran440) [reply hr_holidays_validity_date/models/hr_leave.py]: okay, you are right.

Comment (Saran440) [reply hr_holidays_validity_date/models/hr_leave.py:31]: okay. you are right.

**Review by Saran440 — APPROVED** [review]

**Review by nikul-serpentcs — APPROVED** [review]
Only Code Review LGTM :+1:

**Comment by etobella** [comment]
@jarroyomorales Can you squash translate commits

**Comment by jarroyomorales** [comment]
Done! @etobella

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:

File: hr_holidays_validity_date/migrations/12.0.2.0.0/pre-migration.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    cr = env.cr
+    if openupgrade.column_exists(
+            cr,
+            'hr_leave_type',
+            'date_start'
+    ):
+        openupgrade.logged_query(
+            cr, """
+            UPDATE hr_leave_type
+            SET validity_start = date_start, validity_stop = date_end;
```
Comment (MiquelRForgeFlow) [inline hr_holidays_validity_date/migrations/12.0.2.0.0/pre-migration.py:18]: but:
`validity_start` and `validity_stop` are of type **date**
and `date_start` and `date_end` were of type **datetime**

Comment (AaronHForgeFlow) [reply hr_holidays_validity_date/migrations/12.0.2.0.0/pre-migration.py:18]: Is this an issue?

Comment (jarroyomorales) [reply hr_holidays_validity_date/migrations/12.0.2.0.0/pre-migration.py:18]: The time part will be lost but I dont think its very relevant and there is not much we can do with this anyways.
`date_start` and `date_end` columns are not dropped so if someone really needs it, the information is still there.

Comment (AaronHForgeFlow) [reply hr_holidays_validity_date/migrations/12.0.2.0.0/pre-migration.py:18]: it is ok to me.

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
@jarroyomorales I have conflicts when trying to rebase. Can you check? Otherwise this cannot be merged

**Comment by jarroyomorales** [comment]
@AaronHForgeFlow I just rebased and didnt get any error. Which conflicts are you getting?

**Comment by AaronHForgeFlow** [comment]
opps sorry, I was wrong. Please dismiss my comment, it was my fault.

**Comment by MiquelRForgeFlow** [comment]
please, could you rebase this to fix runbot?

**Comment by jarroyomorales** [comment]
@MiquelRForgeFlow Done!

**Comment by MiquelRForgeFlow** [comment]
ready to merge

**Pull request merged**